ansible_sdk/executors/base.py

```diff
@@ -22,9 +25,14 @@ class AnsibleJobExecutorOptionsBase:
 class AnsibleJobExecutorBase(abc.ABC):
     @abc.abstractmethod
     async def submit_job(self, job_def: AnsibleJobDef, options: AnsibleJobExecutorOptionsBase) -> AnsibleJobStatus:
-        pass
+        if job_def.metrics_output_path and not os.path.exists(job_def.metrics_output_path):
+            os.makedirs(job_def.metrics_output_path)
 
     async def _stream_events(self, reader: asyncio.StreamReader, status_obj: AnsibleJobStatus) -> None:
+        if status_obj._job_def.metrics_output_path:
+            metrics_calc = MetricsCalc()
+            asyncio.create_task(metrics_calc.collect_metrics(status_obj))
```

Comment:
All `Awaitable`s must be `await`ed before exit- we're just dropping this one on the floor. It can be stored as a local and `await`ed after the `while` loop, but to be really safe it probably needs a `try/finally` (since an error here may bubble out and not await it, but also may not be fatal to the process)